File: fizzbuzz.clj
(ns fizzbuzz
  (:require [clojure.string :refer [split split-lines join]]
            [clojure.java.io :refer [reader]]))

(defn get-a-b-n [line]
  (map #(Integer/parseInt % 10) (split line #"\s")))

(defn divisible? [num factor]
  (= (mod num factor) 0))

(defn divide [num a b]
  (cond
    (and (divisible? num a) (divisible? num b)) "FB"
    (divisible? num a) "F"
    (divisible? num b) "B"
    :else (str num)))

(defn process-line [a b n]
  (for [num (range 1 (inc n))]
    (divide num a b)))

(defn -main [& args]
  (let [input-file (nth args 0)]
    (with-open [rdr (reader input-file)]
      (doseq [line (line-seq rdr)]
        (println (join " " (apply process-line (get-a-b-n line))))))))

(apply -main *command-line-args*)

File: fizzbuzz.py
import sys

def divisible(num, factor):
    return num % factor == 0

def divide(num, a, b):
    if divisible(num, a) and divisible(num, b):
        return "FB"
    elif divisible(num, a):
        return "F"
    elif divisible(num, b):
        return "B"
    else:
        return str(num)

def process_line(a, b, n):
    for num in range(1, n+1):
        yield divide(num, a, b)

def main():
    with open(sys.argv[1]) as f:
        for line in f:
            a, b, n = [int(i) for i in line.split()]
            print(" ".join(r for r in process_line(a, b, n)))

if __name__ == '__main__':
    main()
